packages/editor/src/components/unsaved-changes-warning/index.js

```diff
@@ -42,10 +41,4 @@ class UnsavedChangesWarning extends Component {
 	}
 }
 
-export default withSelect( ( select ) => ( {
-	// We need to call the selector directly in the listener to avoid race
-	// conditions with `BrowserURL` where `componentDidUpdate` gets the
-	// new value of `isEditedPostDirty` before this component does,
-	// causing this component to incorrectly think a trashed post is still dirty.
```

Comment:
We should probably retain this comment somewhere to explain why our new `isDirty` is a predicate (rather than a bool) :thinking: Maybe before [R31](https://github.com/WordPress/gutenberg/pull/24659/files#diff-650595694fb11fe3d6310a076c254463R31)?

Reply:
@ockham Gotcha! Restored the comment!